pandas/tests/indexes/base_class/test_reshape.py

```diff
@@ -42,6 +43,17 @@ def test_insert_missing(self, nulls_fixture):
         result = Index(list("abc")).insert(1, nulls_fixture)
         tm.assert_index_equal(result, expected)
 
+    @pytest.mark.parametrize(
+        "val", [(1, 2), np.datetime64("2019-12-31"), np.timedelta64(1, "D")]
+    )
+    @pytest.mark.parametrize("loc", [-1, 2])
+    def test_insert_datetime_into_object(self, loc, val):
+        # GH#44509
+        idx = Index(["1", "2", "3"])
+        result = idx.insert(loc, val)
+        expected = Index(["1", "2", val, "3"])
```

Comment:
can you check that `type(expected[2]) is type(val)`; numpy often messes that up with dt64/td64

Reply:
Like that?

Reply:
https://github.com/numpy/numpy/issues/12550 i dont know exactly what conditions it shows up in, but i get nervous whenever np.datetime64 or np.timedelta64 objects get near an object-dtype ndarray

Reply:
Thx for the reference